### PR TITLE
update packages/multimedia and their dependancies

### PR DESCRIPTION
**Pull request by heitbaum**
Limited testing on Generic so far.

- libass 0.15 now has a mandatory dependancy on harfbuzz
- harfbuzz 2.7.2 if using -Dicu=enable is fixed

all other package updates were quite minor

### CODE REVIEW

**Comment by heitbaum** [comment]
Compile error using -Dwith-icu=enabled when building 2.4.0 or greater https://github.com/harfbuzz/harfbuzz/issues/2769

**Comment by heitbaum** [comment]
Updated with -Dicu=enabled for harfbuzz 
and --enable-renaming for icu

compile error is now resolved. Please review.

**Comment by heitbaum** [comment]
Rebased against #4803

File: packages/multimedia/libass/package.mk [hunk 1]
```diff
@@ -3,17 +3,16 @@
...
 PKG_NAME="libass"
-PKG_VERSION="0.14.0"
-PKG_SHA256="881f2382af48aead75b7a0e02e65d88c5ebd369fe46bc77d9270a94aa8fd38a2"
+PKG_VERSION="0.15.0"
+PKG_SHA256="9f09230c9a0aa68ef7aa6a9e2ab709ca957020f842e52c5b2e52b801a7d9e833"
 PKG_LICENSE="BSD"
 PKG_SITE="https://github.com/libass/libass"
 PKG_URL="https://github.com/libass/libass/releases/download/$PKG_VERSION/$PKG_NAME-$PKG_VERSION.tar.xz"
-PKG_DEPENDS_TARGET="toolchain freetype fontconfig fribidi"
+PKG_DEPENDS_TARGET="toolchain freetype fontconfig fribidi harfbuzz"
```
Comment (lrusak) [inline packages/multimedia/libass/package.mk:11]: I really don't like this but it seems there is no other option. libass is integral to kodi. It's probably worth moving harfbuzz out of the addon depends tree and into packages/graphics or something.

File: packages/multimedia/rtmpdump/package.mk [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 # Copyright (C) 2009-2016 Stephan Raue (stephan@openelec.tv)
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="rtmpdump"
-PKG_VERSION="fa8646daeb19dfd12c181f7d19de708d623704c0"
-PKG_SHA256="dba4d4d2e1c7de6884b01d98194b83cab6784669089fa3c919152087a3a38fd2"
+PKG_VERSION="c5f04a58fc2aeea6296ca7c44ee4734c18401aa3"
```
Comment (lrusak) [inline packages/multimedia/rtmpdump/package.mk:6]: please make sure inputstream.rtmp and curl build with this bump as rtmpdump has been problematic in the past.

Comment (heitbaum) [reply packages/multimedia/rtmpdump/package.mk:6]: Confirmed both 
- https://curl.haxx.se/download/curl-7.74.0.tar.xz
- https://codeload.github.com/xbmc/inputstream.rtmp/tar.gz/3.4.0-Matrix

Building with
- HEAD c5f04a58fc2aeea6296ca7c44ee4734c18401aa3

File: packages/multimedia/libass/package.mk (hunk 1 above)
Comment (heitbaum) [reply packages/multimedia/libass/package.mk:11]: Package moved

File: packages/multimedia/SDL2/patches/SDL2-2.0.12.001.patch [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (lrusak) [inline packages/multimedia/SDL2/patches/SDL2-2.0.12.001.patch:1]: I thought I commented on this but I guess not. Why is this patch needed? Likely we would need SDL2 to support EGL as we enable OpenGL/ES support in SDL2.

Comment (heitbaum) [reply packages/multimedia/SDL2/patches/SDL2-2.0.12.001.patch]: In a further commit I had updated to SDL2 and was able to remove the patch.

**Comment by heitbaum** [comment]
Updated to SDL2: 2.0.14 
- https://www.libsdl.org/tmp/SDL/WhatsNew.txt
- - removed 2.0.12 compile issue patch

**Comment by CvH** [comment]
due the harfbuzz requirement we need to include the 30mb icu lib at the image, can we disable icu at harfbuzz and does it still work ?

File: packages/graphics/harfbuzz/package.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+PKG_NAME="harfbuzz"
+PKG_VERSION="2.7.3"
+PKG_SHA256="97bf10e7bc576a4e60a1cb45388669b778bb7787660d8666c4e554f021cdead9"
+PKG_LICENSE="GPL"
+PKG_SITE="http://www.freedesktop.org/wiki/Software/HarfBuzz"
+PKG_URL="https://github.com/harfbuzz/harfbuzz/releases/download/$PKG_VERSION/harfbuzz-$PKG_VERSION.tar.xz"
+PKG_DEPENDS_TARGET="toolchain cairo freetype glib icu"
+PKG_LONGDESC="HarfBuzz is an OpenType text shaping engine."
+PKG_TOOLCHAIN="meson"
+PKG_DEPENDS_CONFIG="icu"
```
Comment (dhewg) [inline packages/graphics/harfbuzz/package.mk:14]: If that is a hard dependency, icu should be moved out of addon-depends as well with its PKG_BUILD_FLAGS="-sysroot" removed, and then this line can be removed.
That in return allows more cleanups with icu handling in a few packages (`git grep -w icu`).
Note that other packages might detect and use icu then as well.

Comment (heitbaum) [reply packages/graphics/harfbuzz/package.mk:14]: @CvH - yes disabling icu does compile. It was my first attempt at updating harfbuzz. The reason I persisted in figuring out why -Dicu=enable did not work was that icu was enabled previously. Happy either way. Let me know.

Comment (dhewg) [reply packages/graphics/harfbuzz/package.mk:14]: debian has libharfbuzz and libharfbuzz-icu, where only the latter depends on libicu. libass depends on the former, so it should work build a icu less harfbuzz for the base image, and a icu enabled harfbuzz for the addon. If so desired.

Comment (CvH) [reply packages/graphics/harfbuzz/package.mk:14]: yea sounds like plan

Comment (heitbaum) [reply packages/graphics/harfbuzz/package.mk]: Done. harfbuzz renamed to harfbuzz-icu, and new harfbuzz package created.

File: packages/addons/addon-depends/chrome-depends/harfbuzz-icu/package.mk [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # Copyright (C) 2009-2012 Stephan Raue (stephan@openelec.tv)
 # Copyright (C) 2016-present Team LibreELEC (https://libreelec.tv)
 
-PKG_NAME="harfbuzz"
+PKG_NAME="harfbuzz-icu"
```
Comment (CvH) [inline packages/addons/addon-depends/chrome-depends/harfbuzz-icu/package.mk:5]: can we do something like this https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/addons/addon-depends/chrome-depends/chrome-libXcomposite/package.mk ?

so we don't need to maintain two packages

**Review by CvH — APPROVED** [review]

**Pull request merged**